### ISSUE TITLE
docker latest (3.3.7) always returns empty search results

### ISSUE THREAD

**Issue by akatakritos** [microsoft/fhir-server#2923]
**Describe the bug**
In the 3.3.7 (latest) version of the docker container, searches do not seem to return results the same way they do in earlier versions. This is reproducible for me across resource types and search params (Patient, Practitioner, maybe Observation & ServiceResult)

**FHIR Version?**
R4

**Data provider?**
SQL Server

**To Reproduce**

docker-compose.yml 3.3.0 (runs on localhost:8081)
```yml
version: "3.8"
services:

  fhir-database:
    image: mcr.microsoft.com/mssql/server:2019-CU12-ubuntu-20.04
    environment:
      ACCEPT_EULA: "Y"
      MSSQL_SA_PASSWORD: "SuperSecretPassw0rd!"
    healthcheck:
        test: ["CMD", "/opt/mssql-tools/bin/sqlcmd","-U sa -P SuperSecretPassw0rd! -Q 'SELECT * FROM INFORMATION_SCHEMA.TABLES'"]
        interval: 10s
        timeout: 10s
        retries: 6
    ports:
      - 1436:1433
    volumes:
      - fhir-database-volume:/var/opt/mssql

  fhir:
    image: mcr.microsoft.com/healthcareapis/r4-fhir-server:3.3.0
    ports:
      - 8081:8080
    environment:
      FhirServer__Security__Enabled: "false"
      SqlServer__ConnectionString: "Server=tcp:fhir-database,1433;Database=FHIR330;Persist Security Info=False;User ID=sa;Password=SuperSecretPassw0rd!;TrustServerCertificate=True;MultipleActiveResultSets=False;Connection Timeout=30"
      SqlServer__AllowDatabaseCreation: "true"
      SqlServer__Initialize: "true"
      SqlServer__SchemaOptions__AutomaticUpdatesEnabled: "true"
      DataStore: "SqlServer"
    depends_on:
      - fhir-database

volumes:
  fhir-database-volume:
```

docker-compose.yml for 3.3.7 (runs on localhost:8082)
```yml
version: "3.8"
services:

  fhir-database:
    image: mcr.microsoft.com/mssql/server:2019-CU12-ubuntu-20.04
    environment:
      ACCEPT_EULA: "Y"
      MSSQL_SA_PASSWORD: "SuperSecretPassw0rd!"
    healthcheck:
        test: ["CMD", "/opt/mssql-tools/bin/sqlcmd","-U sa -P SuperSecretPassw0rd! -Q 'SELECT * FROM INFORMATION_SCHEMA.TABLES'"]
        interval: 10s
        timeout: 10s
        retries: 6
    ports:
      - 1437:1433
    volumes:
      - fhir-database-volume:/var/opt/mssql

  fhir:
    image: mcr.microsoft.com/healthcareapis/r4-fhir-server:3.3.7
    ports:
      - 8082:8080
    environment:
      FhirServer__Security__Enabled: "false"
      SqlServer__ConnectionString: "Server=tcp:fhir-database,1433;Database=FHIRlatest;Persist Security Info=False;User ID=sa;Password=SuperSecretPassw0rd!;TrustServerCertificate=True;MultipleActiveResultSets=False;Connection Timeout=30"
      SqlServer__AllowDatabaseCreation: "true"
      SqlServer__Initialize: "true"
      SqlServer__SchemaOptions__AutomaticUpdatesEnabled: "true"
      DataStore: "SqlServer"
    depends_on:
      - fhir-database

volumes:
  fhir-database-volume:

```

Put these two files in two different directories, then on two different command lines, one for each folder, run `docker-compose up`

Test Case:

(Using above, port 8081 is running 3.3.0 and 8082 is running 3.3.7)

Create a patient:
```
POST http://localhost:$PORT/Patient
{
  "name": [
    {
      "use": "official",
      "family": "Smith",
      "given": [
        "Darcy"
      ]
    }
  ],
  "gender": "female",
  "birthDate": "1970-01-01",
  "resourceType": "Patient"
}
```

Then search for it

```
GET http://localhost:$PORT/Patient
```

3.3.0:
```json
{
    "resourceType": "Bundle",
    "id": "fb441db1-0497-4cae-88db-14a8ae52b5d7",
    "meta": {
        "lastUpdated": "2022-11-02T12:27:13.8297939+00:00"
    },
    "type": "searchset",
    "link": [
        {
            "relation": "self",
            "url": "http://localhost:8081/Patient"
        }
    ],
    "entry": [
        {
            "fullUrl": "http://localhost:8081/Patient/50dbd069-0332-42ba-8592-193636f25032",
            "resource": {
                "resourceType": "Patient",
                "id": "50dbd069-0332-42ba-8592-193636f25032",
                "meta": {
                    "versionId": "1",
                    "lastUpdated": "2022-11-02T12:27:00.734+00:00"
                },
                "name": [
                    {
                        "use": "official",
                        "family": "Smith",
                        "given": [
                            "Darcy"
                        ]
                    }
                ],
                "gender": "female",
                "birthDate": "1970-01-01"
            },
            "search": {
                "mode": "match"
            }
        }
    ]
}
```

3.3.7
```json
{
    "resourceType": "Bundle",
    "id": "a266cb49-9dc5-4fa4-aba3-afe8997c55df",
    "meta": {
        "lastUpdated": "2022-11-02T12:36:25.7957509+00:00"
    },
    "type": "searchset",
    "link": [
        {
            "relation": "self",
            "url": "http://localhost:8082/Patient"
        }
    ]
}
```

**Expected behavior**
3.3.7 should return all Patients when searching `/Patient`, as 3.3.0 does.

**Actual behavior**
3.3.7 returns an empty search set.

The Patient does exist: `GET /Patient/:id` returns the record

This is reproducible against other queries too, for example`GET /Patient?name=Darcy` works in 3.3.0, fails in 3.3.7

AB#98006

**Comment by brendankowitz** [comment]
Thanks @akatakritos, it looks like this is an issue related to the new middleware that enforces SMART scope access. This middleware isn't checking the Security enabled flag properly.

**Comment by akatakritos** [comment]
Thanks @brendankowitz!

Followup: I'm not too sure what SMART is. We use Azure Health Data Services in PAAS for our deployed environments. Is there configuration we need to do to opt out of SMART? We only authenticate end-users in our app layer, and then use a simple Client Credentials OAuth grant to access the FHIR server (i.e. the webserver authenticates as itself and should be able to read/write anything in the FHIR service; it performs access control for end-users in app code).

**Comment by EXPEkesheth** [comment]
@akatakritos  Issue is fixed with the latest OSS release (3.3.18). For your question, is there configuration we need to do to opt out of SMART? No configuration is needed to opt out of SMART on FHIR, if you have not explicitly selected / configured.
Please let us know if you have additional questions

**Comment by EXPEkesheth** [comment]
Closing the issue
